cypress/e2e/eda/Roles/custom-roles-crud.cy.ts

```diff
@@ -0,0 +1,75 @@
+//Tests a user's ability to create, edit, and delete a custom role in the EDA UI.
+import { randomString } from '../../../../framework/utils/random-string';
+
+describe('EDA Custom Roles- Create', () => {
+  const name = 'E2E Custom Role ' + randomString(4);
+  before(() => {
+    cy.edaLogin();
+  });
+
+  after(() => {
+    cy.navigateTo('eda', 'roles');
+    cy.get('h1').should('contain', 'Roles');
+    cy.clickTableRow(name, true);
+    cy.verifyPageTitle(name);
+    cy.clickPageAction('delete-role');
+    cy.clickModalConfirmCheckbox();
+    cy.clickModalButton('Delete role');
+    cy.verifyPageTitle('Roles');
+  });
+
+  it('can create a custom role', () => {
+    cy.navigateTo('eda', 'roles');
+    cy.get('h1').should('contain', 'Roles');
+    cy.get('[id="create-role"]').click();
+    cy.get('[data-cy="name"]').type(name);
+    cy.get('[data-cy="description"]').type('This is a Custom Role.');
+    cy.get('[data-cy="content-type-form-group"]').click();
+    cy.get('[id="activation"]').click();
+    cy.get('[data-cy="permissions"]').click();
+    cy.get('[data-cy="view-activation"] input').click();
+    cy.clickButton(/^Create role$/);
+    cy.hasDetail('Name', name);
+    cy.hasDetail('Description', 'This is a Custom Role.');
+  });
+});
+
+describe('EDA Custom Roles- Edit & Delete', () => {
+  const name = 'E2E Custom Role ' + randomString(4);
+  before(() => {
+    cy.edaLogin();
+    cy.navigateTo('eda', 'roles');
+    cy.get('h1').should('contain', 'Roles');
+    cy.get('[id="create-role"]').click();
+    cy.get('[data-cy="name"]').type(name);
+    cy.get('[data-cy="description"]').type('This is a Custom Role.');
+    cy.get('[data-cy="content-type-form-group"]').click();
+    cy.get('[id="activation"]').click();
+    cy.get('[data-cy="permissions"]').click();
+    cy.get('[data-cy="view-activation"] input').click();
+    cy.clickButton(/^Create role$/);
+    cy.hasDetail('Name', name);
+  });
+
+  it('can edit a custom role', () => {
+    cy.navigateTo('eda', 'roles');
+    cy.get('h1').should('contain', 'Roles');
+    cy.clickTableRow(name, true);
+    cy.clickButton(/^Edit role$/);
+    cy.verifyPageTitle(`Edit Role`);
+    cy.get('[data-cy="description"]').type('this custom role has been changed');
+    cy.clickButton(/^Save role$/);
+    cy.hasDetail('Description', 'this custom role has been changed');
+  });
+
+  it('can delete a custom role', () => {
```

Comment:
Hi Apurva! I ran the tests on this spec, and saw the following error with this test:
![image](https://github.com/ansible/ansible-ui/assets/63685497/3f366809-09fe-4da8-abca-15e88cf6204e)